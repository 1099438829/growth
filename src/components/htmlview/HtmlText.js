import React, { Component, PropTypes } from 'react';
import { Text } from 'react-native';

function getTextStyle(node) {
  return (
  node.parent &&
  node.parent.parent &&
  /blockquote|code/.test(node.parent.parent.name))
    ? { color: '#4c4c4c' }
    : { color: '#000' };
}

class HtmlText extends Component {
  static componentName = 'HtmlText';

  static propTypes = {
    text: PropTypes.string,
    component: PropTypes.node,
    parentName: PropTypes.string,
  };

  static defaultProps = {
    text: '',
    component: null,
    parentName: '',
  };
  render() {
    return (<Text
      style={getTextStyle(this.props.parentName)}
    >
      {this.props.text}
      <Text>{this.props.component}</Text>
    </Text>);
  }
}
export default HtmlText;
