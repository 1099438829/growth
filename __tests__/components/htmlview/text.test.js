import { Text } from 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import HtmlText from '../../../src/components/htmlview/HtmlText';

it('renders correctly', () => {
  // eslint-disable-next-line no-unused-vars
  const text = renderer.create(
    <HtmlText text={2333} component={<Text>2333</Text>} />,
  );
  const textJson = text.toJSON();
  expect(textJson).toMatchSnapshot();
});
