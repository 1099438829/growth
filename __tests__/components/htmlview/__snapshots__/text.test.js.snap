// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#000",
    }
  }
>
  2333
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      2333
    </Text>
  </Text>
</Text>
`;
